trailing_spaces.py

```diff
@@ -471,15 +465,12 @@ def freeze_last_version(self, view):
         if file_name and not view.is_scratch() and isfile(file_name):
             encoding = view.encoding()
 
-            if encoding == "Undefined":
-                encoding = view.settings().get("default_encoding", "UTF-8")
-
             if encoding == "Hexadecimal":  # not supported?
                 on_disk = None
                 return
 
-            if "(" in encoding:
-                encoding = encoding.split("(")[1].split(")")[0]
+            match = re.match(r'.+\(([^)]+)\)$', view.encoding())
+            encoding = match.group(1) if match else view.settings().get("default_encoding", "UTF-8")
```

Comment:
This would replace "UTF-8", "UTF-16 LE" and "UTF-16 BE" with the "default_encoding"
and would fail to extract from brackets in the "default_encoding".

Reply:
Can you explain the issue more?

Are you saying that `default_encoding` can be in a format like `Central European (ISO 8859-2)`?

Reply:
Yes

Reply:
Right, it seems like it can be. OK, the change looks good to me. Just moved the "if" down which is just a style choice.

Reply:
Not if `default_encoding` is `Hexadecimal`